pkgs/os-specific/linux/kernel/patches.nix

```diff
@@ -59,4 +59,16 @@ rec {
     };
   };
 
+  # https://github.com/NixOS/nixpkgs/issues/42755
+  xen-netfront_fix_mismatched_rtnl_unlock = {
+    name = "xen-netfront_fix_mismatched_rtnl_unlock";
+    patch = ./xen-netfront_fix_mismatched_rtnl_unlock.patch;
```

Comment:
Instead of checking in the patches to nixpkgs you should use `fetchpatch`:
```nix
patch = fetchpatch {
  name = "xen-netfront_fix_mismatched_rtnl_unlock.patch";
  url = "https://github.com/torvalds/linux/commit/cb257783c2927b73614b20f915a91ff78aa6f3e8.patch";
  sha256 = "1288a5irvq2rhfl0aclb0lvhxwmbqp5blcdpx7d7lwz356cwx8xm";
};
```